### PR TITLE
Feature/touchstone rewrite

### PR DESCRIPTION
**Pull request by zarath**
Rewrite of touchstone parser and fix for bug #99

### CODE REVIEW

**Comment by zarath** [comment]
Not ready yet

File: NanoVNASaver/NanoVNASaver.py [hunk 1]
```diff
@@ -905,6 +905,7 @@ def loadSweepFile(self):
             self.data21 = []
             t = Touchstone(filename)
             t.load()
+            # shouldn't modify read file. even destroys it if read fails
```
Comment (mihtjel) [inline NanoVNASaver/NanoVNASaver.py]: It doesn't, so this comment should be removed. NanoVNASaver.saveData saves the data to the application, and does not touch the files.

File: NanoVNASaver/Touchstone.py [hunk 2]
```diff
@@ -13,132 +14,160 @@
...
+            elif p in "syzgh" and not pparam:
+                self.parameter = p
+                pparam = True
+            elif p in ("ma", "db", "ri") and not pformat:
+                self.format = p
+                pformat = True
+            elif p == "r" and not presist:
+                self.resistance = int(next(params))
+            else:
+                raise TypeError("Illegial option line: " + line)
```
Comment (mihtjel) [inline NanoVNASaver/Touchstone.py]: "Illegal" - though better word choice would be "Invalid" :-)

File: NanoVNASaver/Touchstone.py [hunk 3]
```diff
@@ -13,132 +14,160 @@
...
+
+                # ignore comments at data end
+                data = line.split('!')[0]
+                data = data.split()
+                freq, data = float(data[0]) * self.opts.factor, data[1:]
+                data_len = len(data)
+
+                # consistency checks
+                if freq <= prev_freq:
+                    raise TypeError("Frequeny not ascending: " + line)
```
Comment (mihtjel) [inline NanoVNASaver/Touchstone.py]: "Frequency"

**Comment by mihtjel** [comment]
Ready for merge?

**Comment by zarath** [comment]
I think, now it's ready to merge,

I've overseen that phase information in touchstone is in degrees, so it has to be converted to radians before creating a complex from polar

**Pull request merged**